luxonis_ml/data/datasets/annotation.py

```diff
@@ -2,142 +2,151 @@
-    @type class_: str
-    @ivar class_: The class name for the annotation.
-    @type instance_id: int
-    @ivar instance_id: The instance id of the annotation. This
-        determines the order in which individual instances are loaded in
-        L{luxonis_ml.data.LuxonisLoader}.
-    @type _label_type: ClassVar[LabelType]
-    @ivar _label_type: The label type of the annotation.
-    """
+    raise ValueError(f"Unknown label type: {label_type}")
 
-    _label_type: ClassVar[LabelType]
 
-    task: str
-    class_: str = Field("", alias="class")
+class Detection(BaseModelExtraForbid):
+    class_name: Optional[str] = Field(
+        None, validation_alias=AliasChoices("class", "class_name")
+    )
     instance_id: int = -1
 
-    @model_validator(mode="before")
-    @classmethod
-    def validate_task(cls, values: Dict[str, Any]) -> Dict[str, Any]:
-        if "task" not in values:
-            values = {**values, "task": cls._label_type.value}
-        return values
-
-    def get_value(self) -> Dict[str, Any]:
-        """Converts the annotation to a dictionary that can be saved to
-        a parquet file."""
-        return self.dict(
-            exclude={"class_", "class_id", "instance_id", "task", "type_"}
+    metadata: Dict[str, Union[int, float, str]] = {}
+
+    boundingbox: Optional["BBoxAnnotation"] = None
+    keypoints: Optional["KeypointAnnotation"] = None
+    segmentation: Optional["SegmentationAnnotation"] = None
+    array: Optional["ArrayAnnotation"] = None
+
+    scale_to_boxes: bool = False
+
+    sub_detections: Dict[str, "Detection"] = {}
+
+    def to_parquet_rows(self, prefix: str = "") -> Iterable[ParquetDetection]:
+        for label_type in [
+            "boundingbox",
+            "keypoints",
+            "segmentation",
+            "array",
+        ]:
+            label: Optional[Annotation] = getattr(self, label_type)
+            if label is not None:
+                yield {
+                    "class_name": self.class_name,
+                    "instance_id": self.instance_id,
+                    "label_type": f"{prefix}{label_type}",
+                    "annotation": label.model_dump_json(),
+                }
+        for key, data in self.metadata.items():
+            yield {
+                "class_name": self.class_name,
+                "instance_id": self.instance_id,
+                "label_type": f"{prefix}metadata/{key}",
+                "annotation": json.dumps(data),
+            }
+        if self.class_name is not None:
+            yield {
+                "class_name": self.class_name,
+                "instance_id": self.instance_id,
+                "label_type": f"{prefix}classification",
+                "annotation": "{}",
+            }
+        for name, detection in self.sub_detections.items():
+            yield from detection.to_parquet_rows(f"{prefix}{name}/")
+
+    @model_validator(mode="after")
+    def rescale_values(self) -> Self:
```

Comment:
When is this used?

Reply:
It's used if you are using nested labels and want to pass for example the keypoints using coordinates relative to the bbox and not the entire image. I'm not sure though if there's ever going to be a use case for this, so maybe we can just remove this option.

Reply:
I see. I guess we can keep then for this option to be available